src/test/java/jenkins/advancedqueue/jobinclusion/strategy/JobInclusionJobPropertyTest.java

```diff
@@ -3,53 +3,238 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
+import hudson.Launcher;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.StreamBuildListener;
+import hudson.util.ListBoxModel;
+import hudson.util.StreamTaskListener;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import jenkins.advancedqueue.PriorityConfiguration;
+import org.junit.After;
 import org.junit.Before;
+import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TestName;
 import org.jvnet.hudson.test.JenkinsRule;
 
 public class JobInclusionJobPropertyTest {
-    private JobInclusionJobProperty jobProperty;
-    private JobInclusionJobProperty.DescriptorImpl descriptor;
+
+    @ClassRule
+    public static JenkinsRule j = new JenkinsRule();
 
     @Rule
-    public JenkinsRule jenkinsRule = new JenkinsRule();
+    public TestName testName = new TestName();
+
+    private JobInclusionJobProperty property;
+    private FreeStyleProject project;
+    private FreeStyleBuild build;
+    private StreamBuildListener listener;
+    private Launcher launcher;
 
     @Before
-    public void setup() {
-        jobProperty = new JobInclusionJobProperty(true, "TestJobGroup");
-        descriptor = new JobInclusionJobProperty.DescriptorImpl();
+    public void setUp() throws Exception {
+        property = new JobInclusionJobProperty(true, "testGroup");
```

Comment:
Please reduce this difference to the existing test class, since changing the name of the property does not help the test and it makes the review of differences more difficult.
```suggestion
        jobProperty = new JobInclusionJobProperty(true, "TestJobGroup");
```